packages/aws-cdk-lib/aws-elasticache/lib/elasticache-cluster.ts

```diff
@@ -0,0 +1,139 @@
+   */
+  readonly parameterGroupFamily?: string;
+  
+  /**
+   * Security groups to attach to the cluster
+   * @default - A new security group is created
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+  
+  /**
+   * The weekly time range during which maintenance updates can occur.
+   * @default - No preference
+   */
+  readonly preferredMaintenanceWindow?: string;
+  
+  /**
+   * Specifies the weekly time range during which maintenance 
+   * on the cache cluster is performed
+   */
+  readonly port?: number;
+}
+
+export class ElastiCacheCluster extends Construct {
+  /**
+   * The underlying ElastiCache cluster resource
+   */
+  public readonly cluster: elasticache.CfnCacheCluster;
+  
+  /**
+   * The security group associated with the cluster
+   */
+  public readonly securityGroup: ec2.ISecurityGroup;
+  
+  /**
+   * The subnet group where the cluster is placed
+   */
+  public readonly subnetGroup: elasticache.CfnSubnetGroup;
+
+  constructor(scope: Construct, id: string, props: ElastiCacheClusterProps) {
+    super(scope, id);
+
+    // Create a subnet group for the cluster
+    this.subnetGroup = new elasticache.CfnSubnetGroup(this, 'SubnetGroup', {
+      description: 'Subnet group for ElastiCache cluster',
+      subnetIds: props.vpc.privateSubnets.map(subnet => subnet.subnetId),
+    });
+
+    // Create or use provided security group
+    if (props.securityGroups && props.securityGroups.length > 0) {
+      this.securityGroup = props.securityGroups[0];
+    } else {
+      this.securityGroup = new ec2.SecurityGroup(this, 'SecurityGroup', {
+        vpc: props.vpc,
+        description: 'Security group for ElastiCache cluster',
+        allowAllOutbound: true,
+      });
+    }
+
+    // Create parameter group if family is specified
+    let parameterGroup;
+    if (props.parameterGroupFamily) {
+      parameterGroup = new elasticache.CfnParameterGroup(this, 'ParameterGroup', {
+        cacheParameterGroupFamily: props.parameterGroupFamily,
+        description: 'Parameter group for ElastiCache cluster',
+      });
+    }
+
+    // Create the ElastiCache cluster
+    this.cluster = new elasticache.CfnCacheCluster(this, 'Resource', {
+      engine: 'redis',
+      cacheNodeType: props.nodeType ?? 'cache.t3.micro',
+      numCacheNodes: props.numCacheNodes ?? 1,
+      engineVersion: props.engineVersion ?? '6.x',
+      vpcSecurityGroupIds: [this.securityGroup.securityGroupId],
+      cacheSubnetGroupName: this.subnetGroup.ref,
+      ...(parameterGroup && { cacheParameterGroupName: parameterGroup.ref }),
+      ...(props.preferredMaintenanceWindow && {
+        preferredMaintenanceWindow: props.preferredMaintenanceWindow,
+      }),
+      ...(props.port && { port: props.port }),
+    });
+  }
+
+  /**
+   * Allow incoming connections on the default Redis port
+   */
+  public allowIngressFrom(peer: ec2.IPeer, port?: number) {
+    this.securityGroup.addIngressRule(
+      peer,
+      ec2.Port.tcp(port ?? 6379),
```

Comment:
it added the default port here, but it did not use it while creating the `CfnCacheCluster` resource